babel/messages/frontend.py

```diff
@@ -852,6 +853,247 @@ def run(self):
+                sort_output=self.sort_output,
+                no_location=self.no_location,
+            )
+
+
+class MergeCatalog(CommandMixin):
+    description='updates translation PO file by merging them with updated template POT file with using compendium'
+    user_options=[
+        ('input-files', None, 'def.po (obsolete translations) ref.pot (actual template)'),
+        ('compendium=', 'C', 'additional library of message translations, may be specified more than once'),
+        ('compendium-overwrite', '', 'overwrite mode of compendium'),
+        ('no-compendium-comment', '', ''),
+        ('update', 'U', 'pdate def.po, do nothing if def.po already up to date'),
+        ('output-file=', 'o', 'write output to specified file, the results are written '
+                              'to standard output if no output file is specified'),
+        ('backup', None, 'make a backup of def.po'),
+        ('suffix=', None, 'override the usual backup suffix'),
+        ('no-fuzzy-matching', 'N', 'do not use fuzzy matching'),
+        ('no-location', None, 'suppress \'#: filename:line\' lines'),
+        ('width=', 'w', 'set output page width'),
+        ('no-wrap', None, 'do not break long message lines, longer '
+                          'than the output page width, into several lines'),
+        ('sort-output', 's', 'generate sorted output'),
+        ('sort-by-file', 'F', 'sort output by file location'),
+    ]
+
+    as_args = 'input-files'
+
+    multiple_value_options = (
+        'compendium'
+    )
+
+    boolean_options = [
+        'compendium-overwrite',
+        'no-compendium-comment',
+        'update',
+        'backup',
+        'no-fuzzy-matching',
+        'no-location',
+        'no-wrap',
+        'sort-output',
+        'sort-by-file',
+    ]
+
+    def initialize_options(self):
+        self.input_files = None
+        self.compendium = None
+        self.compendium_overwrite = False
+        self.no_compendium_comment = False
+        self.update = False
+        self.output_file = None
+        self.backup = False
+        self.suffix = '~'
+        self.no_fuzzy_matching = False
+        self.no_location = False
+        self.width = None
+        self.no_wrap = False
+        self.sort_output = False
+        self.sort_by_file = False
+
+    def finalize_options(self):
+        if not self.input_files or len(self.input_files) != 2:
+            raise OptionError('must be two po files')
+        if not self.output_file and not self.update:
+            raise OptionError('you must specify the output file or update existing')
+
+        if self.no_wrap and self.width:
+            raise OptionError("'--no-wrap' and '--width' are mutually exclusive")
+        if not self.no_wrap and not self.width:
+            self.width = 76
+        elif self.width is not None:
+            self.width = int(self.width)
+
+    def _get_message_from_compendium(self, compendium):
+        for file_path in compendium:
```

Comment:
Even if the CLI argument is in the singular, it would be nicer to read here in the plural.

Also, this clearly returns things in the plural:

```suggestion
    def _get_messages_from_compendiums(self, compendium_paths):
        if not compendium_paths:
            return
        for file_path in compendium_paths:
```